File: 0287. Find the Duplicate Number/sort.cpp
class Solution {
public:
    int findDuplicate(vector<int>& v) {
        if(v.size() == 0)
            return 0;
        sort(v.begin(), v.end());
        int len = v.size(), last = v[0];
        for(int i = 1; i < len; ++i){
            if(v[i] != last){
                last = v[i];
            }else{
                return last;
            }
        }
        return -1;
    }
};

File: 0287. Find the Duplicate Number/sort.java
class Solution {
    public int findDuplicate(int[] v) {
        Arrays.sort(v);
        int len = v.length, index = 1, lastVal = v[0];
        while(index < len){
            if(v[index] == lastVal){
                return v[index];
            }else{
                lastVal = v[index];
                index++;
            }
        }
        return -1;
    }
}